enhancements/network/kubernetes-nmstate-operator-to-cvo.md

```diff
@@ -0,0 +1,279 @@
+
+For externalized controlplane deployments this would be needed for both the management cluster and the externalized controlplane, so that the configuration of NICs can be handled in a consistent way - it’s an open question how we’d handle this in the hybrid platform case though...
+
+
+**Q:** is it ok to deliver the content in a release payload, but enable it via some other configuration knob?
+
+**A:** The desire is to enable a common enabled-by-default API for network configuration (initially only for selected platforms where that is commonly needed).
+
+If that approach is not acceptable then delivering via the release payload with some way to opt-in would be reasonable.
+
+
+**Q:** What is the knob to turn on/off the capability? This is related to the SLO chosen to deliver this new meta-operator and what configuration it reads to know to turn it on. For example, if this is for selected platforms, the component would read the Infrastructure CR to determine if it should enable it all, but is there another knob that would allow opt-in/out?
+
+**A:** The current proposal is to couple this to the platform detected in the Infrastructure CR, so it can be enabled by default on selected platforms (particularly baremetal).
+
+There could also be some other interface at the SLO level to override that behavior and/or to enable opt-in for platforms where it’s not enabled by default.
+
+
+### Goals
+
+- Provide NMState APIs by default on desired platforms, [NetworkManager configuration should be updated](https://github.com/openshift/enhancements/blob/master/enhancements/machine-config/mco-network-configuration.md#option-c-design) to support this capability on these platforms.
+- Allow using NMState APIs at install-time, where NIC config is a common requirement.
+
+### Non-Goals
+
+- Make persistent networking changes to nodes. That should be handled by Machine Config Operator ( check [this link](https://github.com/openshift/enhancements/blob/master/enhancements/machine-config/mco-network-configuration.md#option-c) for more details).
+- Provide an API for controlplane network configuration via openshift-install (although in future a common API would be desirable for controlplane and secondary network interfaces)
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As a user of Baremetal IPI I want to provide NodeNetworkConfigurationPolicy resources via manifests at install-time, to simplify
+my deployment workflow and avoid additional post-deploy steps.
+
+I'd like to avoid dependencies on additional registries in disconnected environments and enable core platform functions
+such as network configuration directly from the release payload.
+
+#### Story 2
+
+As a user of Openshift Virtualization I want to apply a NodeNetworkConfigurationPolicy to create network resources at nodes that will connet to VMs nics and use the already deployed kubernetes-nmstate version from openshift to reduce deployment complexity and maintenance.
+
```

Comment:
@bengland2 mentioned OCS as a potential user of this, for configuration of MTU/bonding when secondary networks are used for storage traffic, @yboaron can we capture that here as another user-story?

Reply:
Sure,
Thanks